src/parser/parser.yy

```diff
@@ -555,6 +555,9 @@ expression
     }
     | compound_expression {
         $$ = $1;
+        if(!$$->checkDepth()){
+            throw nebula::GraphParser::syntax_error(@1, "The above expression is not a valid expression, because its depth exceeds the maximum depth!");
```

Comment:
Although the expression exceeds the maximum depth, it's still is a valid expression in theory.  So it's better to adjust the error message.